File: ruby/easy/find_smallest_letter_greater_than_target.rb
# @param {Character[]} letters
# @param {Character} target
# @return {Character}
def next_greatest_letter(letters, target)
    start, stop = 0, letters.length-1
    while start <= stop
        middle = (start+stop)/2
        if(letters[middle].ord > target.ord)
            stop = middle-1
        else
            start = middle+1
        end
    end
    return (start >= letters.length) ? (letters[0]) : letters[start]
end

File: javascript/easy/findSmallestLetterGreaterThanTarget.js
/**
 * @param {character[]} letters
 * @param {character} target
 * @return {character}
 */
var nextGreatestLetter = function(letters, target) {
    let start = 0, end = letters.length-1;
    while(start <= end) {
        let middle = parseInt((start + end)/2);
        if(letters[middle] > target) {
            end = middle - 1;
        } else {
            start = middle + 1;
        }
    }
    return (start >= letters.length) ? letters[0] : letters[start];
};
